framework/src/application.ts

```diff
@@ -285,6 +289,14 @@ export class Application {
 	// --------------------------------------
 	// Private
 	// --------------------------------------
+
+	private _registerDefaultModule(Module: typeof BaseModule): void {
+		assert(Module, 'Module implementation is required');
+		const InstantiableModule = Module as InstantiableBaseModule;
+		const moduleInstance = new InstantiableModule(this.config.genesisConfig);
+		this._node.registerModule(moduleInstance);
+	}
```

Comment:
Can we just call it `_registerModule` and have extra parameter `validate id: boolean` or `default: boolean` . Then we can use it inside the `registerModule` as well within the class, to reuse the logic. This will help in future if there is nay update in that logic.